### ISSUE TITLE
Support for tracking CSS animation position changes

### ISSUE THREAD

**Issue by AndrewShike** [niklasvh/html2canvas#1168]
So I'm working on a project that uses this library for the perhaps unconventional task of capturing individual image frames from a CSS animation. From my tests it looks like html2canvas gets position/sizing data from the element style, but I know I can get that same information in a "live" way by using `getBoundingClientRect()`.

I'm willing to do the work to fork and add in a new option for this, just wondering if someone wouldn't mind giving me some direction on where to look in code and how manageable this task might be.

Thanks in advance x

**Comment by eKoopmans** [comment]
Hey @AndrewShike, sounds like fun! I'm pretty sure `getBoundingClientRect()` is already used to get element bounds, you can see it at work in `getBounds()` (in `src/utils.js`), which is called by `NodeParser.parseBounds()`.

It caches the value into `this.bounds` to reduce computation, so maybe that's the problem you're facing, but I think there's another more likely problem. All elements are first cloned into an iframe before being rendered, so any CSS transitions etc will be reset to 0, they'll be brand new elements. Without knowing more about what you're doing and how though, that's only my best guess!

**Comment by AndrewShike** [comment]
Hmph, ok that makes more sense. I had a think about this and I believe I can find a workaround on my side by querying the CSS transitions and transforming them into javascript animations using `element.animate()` (which should be pretty trivial with how the [Web Animations API works](https://developer.mozilla.org/en-US/docs/Web/API/Web_Animations_API)). 

That way each time a fame is captured I'm then able to `.pause()` every animation (which would resolve any rendering time issues), then compute the values for the animated attributes based on the `animation.currentTime` and the specified [easing](https://developer.mozilla.org/en-US/docs/Web/API/AnimationEffectTimingProperties/easing) function, then write the computed attribute values back into CSS so html2canvas can read it the way it already does. Kinda counter-intuitive but it's probably going to be easier than forking a large library I just found. I made a simple proof of concept and it seems like it's going to work out well.

What I'm making by the way is a live paste bin style web tool specifically for making short animations, with an option to render to video (leveraging [`fluent-ffmpeg`](https://github.com/fluent-ffmpeg/node-fluent-ffmpeg))

Thanks anyway for the tip though!
